chips/sam4l/src/adc.rs

```diff
@@ -332,9 +354,21 @@ impl hil::adc::Adc for Adc {
                     return ReturnCode::FAIL;
                 }
             }
+
+            ReturnCode::SUCCESS
         }
+    }
+}
 
-        ReturnCode::SUCCESS
+/// Implements an ADC capable reading ADC samples on any channel.
+impl hil::adc::Adc for Adc {
+    type Channel = AdcChannel;
+
+    /// Enable and configure the ADC.
+    /// This can be called multiple times with no side effects.
+    fn initialize(&self) -> ReturnCode {
```

Comment:
A HIL is intended to be hardware independent; just because one implementation has an empty init() doesn't mean all will. It seems very safe and prudent to keep init() in there.